api/src/deployment.rs

```diff
@@ -485,6 +485,10 @@ impl DeploymentSystem {
                 let mut lock = deployment.state.write().await;
                 if let DeploymentState::Deployed(DeployedState { so, handle, .. }) = lock.take() {
                     handle.abort();
+                    handle
+                        .await
+                        .map_err(|e| DeploymentApiError::Internal(e.to_string()))?
```

Comment:
If you abort and then `map_err(...)?` the `JoinHandle` will result in an `Err(JoinError)` since the task is cancelled no?

[Awaiting a [cancelled](https://docs.rs/tokio/latest/tokio/task/struct.JoinError.html#method.is_cancelled) task might complete as usual if the task was already completed at the time it was cancelled, but most likely it will fail with a cancelled JoinError.](https://docs.rs/tokio/latest/tokio/task/struct.JoinHandle.html#impl).

This is fine but the issue is the `?` which will bubble the error up and return an API error. Am I missing something here?

Reply:
You also need to apply this diff in the imports (line 18):

```
-use shuttle_service::loader::{Loader, ServeHandle};
+use shuttle_service::loader::Loader;
+use shuttle_service::ServeHandle;
```

otherwise the code doesn't compile

Reply:
Good catch, I missed the `JoinError::Cancelled`

Reply:
Good catch, I missed the `JoinError::Cancelled`